### PR TITLE
Mev/address pr 33 comments

### PR DESCRIPTION
**Pull request by mikivee**
Addressing unresolved comments from #33 (for [RAT-291](https://rewiringamerica.atlassian.net/browse/RAT-291?atlOrigin=eyJpIjoiZDJlZDdlN2U1ZmQyNDJkNWI0YzQ5ODFjYzJmNmYyZjYiLCJwIjoiaiJ9))which I did in a separate PR to minimize merge conflicts with #34 . 

[RAT-291]: https://rewiringamerica.atlassian.net/browse/RAT-291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/datagen.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ class DataGenerator(tf.keras.utils.Sequence):
     building_feature_table_name = get_most_recent_table_version("ml.surrogate_model.building_features")
     weather_feature_table_name = get_most_recent_table_version("ml.surrogate_model.weather_features_hourly")
     target_table_name = get_most_recent_table_version("ml.surrogate_model.building_simulation_outputs_annual")
-    # load the default features, targets, and upgrades to use for this training run based on params stored in current version's config
-    data_params = read_json(LOCAL_ARTIFACT_PATH / CURRENT_VERSION_NUM / "features_targets_upgrades.json")
```
Comment (mikivee) [inline src/datagen.py:55]: Addressing [this comment](https://github.com/rewiringamerica/surrogate_modeling/pull/33#discussion_r1999917728)

File: src/globals.py [hunk 2]
```diff
@@ -11,3 +11,20 @@
...
+# Processed Data Table Paths
+BUILDING_METADATA_TABLE = f'{CATALOG}.{SUMO_DB}.building_metadata'
+MEGASTOCK_BUILDING_METADATA_TABLE = f'{CATALOG}.{MEGASTOCK_DB}.building_metadata'
+ANNUAL_OUTPUTS_TABLE = f'{CATALOG}.{SUMO_DB}.building_simulation_outputs_annual'
+WEATHER_DATA_TABLE = f'{CATALOG}.{SUMO_DB}.weather_data_hourly'
+
+# Feature Table Paths
+BUILDING_FEATURE_TABLE = f'{CATALOG}.{SUMO_DB}.building_features' #NOTE: contains rows for all upgrades 
+MEGASTOCK_BUILDING_FEATURE_TABLE = f'{CATALOG}.{MEGASTOCK_DB}.building_features' #NOTE: containes rows for baseline only
+WEATHER_FEATURE_TABLE = f'{CATALOG}.{SUMO_DB}.weather_features_hourly'
```
Comment (mikivee) [inline src/globals.py]: Addressing [this ](https://github.com/rewiringamerica/surrogate_modeling/pull/33#discussion_r1999972473)PR comment. Note that most of the files changed in this PR are just the downstream effects of importing and using these.

File: model_artifacts/features_targets_upgrades.json
Comment (SkittyWitty) [inline model_artifacts/features_targets_upgrades.json:1]: Just to review for my understanding. The version has been removed and now this is the most recent version of the features, targerts and upgrade configuration. Then when the model is changed will save this off as version 01.00.01 and increment to 01.00.02?

File: model_artifacts/readme.md [hunk 3]
```diff
@@ -4,24 +4,24 @@ This directory stores model outputs, including trained model metadata and evalua
...
+│   ├── mappings.json
 │   ├── features_targets_upgrades.json 
 │   ├── metrics_by_upgrade_type.csv
 ```
 
 ## Versioning
 - Each subdirectory corresponds to a specific model version (e.g., `01_00_00`), which correspond to the poetry version of this repo package, as in `pyproject.toml`.
 - The versioning follows `major_minor_patch` format.
-- Artifacts are stored in Databricks and synchronized here for tracking.
+- The current version of the `features_targets_upgrades.json` and `metrics_by_upgrade_type.csv` are stored in the top level `model_artifacts/` folder in git in order to track paramter and result changes alongside code changes.
```
Comment (SkittyWitty) [inline model_artifacts/readme.md:23]: To review. The version folder has been removed and now the `model_artifacts` directory contains the current versions of the model configurations. Then when the model is changed the configs are versioned and saved to GCS? Could we update the name of `model_artifacts` to be say `model_configs` since all artifacts are now living in GCS?

```suggestion
- The current version of the `features_targets_upgrades.json` and `metrics_by_upgrade_type.csv` are stored in the top level `model_artifacts/` folder in git in order to track parameter and result changes alongside code changes.
```

File: scripts/build_feature_store_02.py [hunk 4]
```diff
@@ -85,8 +85,12 @@
 # COMMAND ----------
 
 # DBTITLE 1,Drop rows where upgrade was not applied
+# get most recent table version for annual outputs to compare against
+outputs_most_recent_version_num = versioning.get_most_recent_table_version(
```
Comment (SkittyWitty) [inline scripts/build_feature_store_02.py]: Nice to have: alternative to the `return_version_number_only` parameter is to create an object with version number and the full table name.

File: scripts/model_evaluation_04.py [hunk 5]
```diff
@@ -99,7 +99,11 @@
 # run inference: output a N x M matrix of predictions
 # where N is the number of rows in the input data table
 # and M is the number of target columns
+<<<<<<< HEAD
```
Comment (SkittyWitty) [inline scripts/model_evaluation_04.py]: Merge conflict to be resolved

File: src/surrogate_model.py [hunk 6]
```diff
@@ -13,7 +13,7 @@
 from tensorflow.keras import layers, models
 from tensorflow.io import gfile
 
-from src.globals import GCS_ARTIFACT_PATH
+from src.globals import GCS_ARTIFACT_PATH, CURRENT_VERSION_NUM
 from src.datagen import DataGenerator
 from src.versioning import get_poetry_version_no
```
Comment (SkittyWitty) [inline src/surrogate_model.py:18]: Is this still being used? I think the `CURRENT_VERSION_NO` covers this.

File: tests/test_feature_utils.py [hunk 7]
```diff
@@ -56,8 +56,10 @@ class ApplyUpgrades(unittest.TestCase):
     )
     def test_apply_upgrades(self):
         """Test feautre upgrade tranformations match expected output."""
-        baseline_features = read_test_baseline_inputs()
-        df_out_expected = read_test_upgraded_outputs()
+        # baseline_features = read_test_baseline_inputs()
```
Comment (SkittyWitty) [inline tests/test_feature_utils.py]: Q: any particular reason to comment instead of removing?

**Review by SkittyWitty — CHANGES_REQUESTED** [review]
Minor changes needed surrounding a merge conflict and possibly unused code. The rest are suggestions. Thanks for administering to my PR comments!

File: model_artifacts/readme.md (hunk 3 above)
Comment (mikivee) [reply model_artifacts/readme.md]: it also contains results though, like a summary performance on a held out test set. Thats not a config 🤔

File: scripts/model_evaluation_04.py (hunk 5 above)
Comment (mikivee) [reply scripts/model_evaluation_04.py]: I think you're not looking at all the commits-- this was already fixed/

File: tests/test_feature_utils.py (hunk 7 above)
Comment (mikivee) [reply tests/test_feature_utils.py]: hmm i think something went wonky when it switched the base branch.. these comment out lines are also not in this branch or in main.

File: scripts/build_feature_store_02.py (hunk 4 above)
Comment (SkittyWitty) [reply scripts/build_feature_store_02.py]: totally understandable.

File: src/globals.py (hunk 2 above)
Comment (SkittyWitty) [reply src/globals.py]: Yay!!! I love seeing these all together like this much easier to keep track for someone digging into the code and hopefully when working with the notebooks too.

File: model_artifacts/readme.md (hunk 3 above)
Comment (SkittyWitty) [reply model_artifacts/readme.md]: I had originally thought we were moving away from the versioned folders. But our last discussion I think we reviewed why we would keep some items with the repo itself. In this case the naming makes sense.

**Review by SkittyWitty — APPROVED** [review]

File: src/globals.py (hunk 2 above)
Comment (mikivee) [reply src/globals.py]: totally agree!! It was a great suggestion
